src/main/java/org/kohsuke/github/GHRepositoryRule.java

```diff
@@ -0,0 +1,667 @@
+            super(key);
+        }
+
+        @Override
+        TypeReference<Boolean> getType() {
+            return new TypeReference<Boolean>() {
+            };
+        }
+    }
+
+    /**
+     * Integer parameter for a ruleset.
+     */
+    public static class IntegerParameter extends Parameter<Integer> {
+        /**
+         * Instantiates a new integer parameter.
+         *
+         * @param key
+         *            the key
+         */
+        public IntegerParameter(String key) {
+            super(key);
+        }
+
+        @Override
+        TypeReference<Integer> getType() {
+            return new TypeReference<Integer>() {
+            };
+        }
+    }
+
+    /**
+     * List parameter for a ruleset.
+     *
+     * @param <T>
+     *            the type of the list
+     */
+    public abstract static class ListParameter<T> extends Parameter<List<T>> {
+        /**
+         * Instantiates a new list parameter.
+         *
+         * @param key
+         *            the key
+         */
+        public ListParameter(String key) {
+            super(key);
+        }
+    }
+
+    /**
+     * Status check configuration parameter.
+     */
+    public static class StatusCheckConfiguration {
+        private String context;
+        private Integer integrationId;
+
+        /**
+         * Instantiates a new status check configuration.
+         */
+        public StatusCheckConfiguration() {
+        }
+
+        /**
+         * Instantiates a new status check configuration.
+         *
+         * @param context
+         *            the context
+         * @param integrationId
+         *            the integration id
+         */
+        public StatusCheckConfiguration(String context, Integer integrationId) {
+            this.context = context;
+            this.integrationId = integrationId;
+        }
+
+        /**
+         * Gets the context.
+         *
+         * @return the context
+         */
+        public String getContext() {
+            return this.context;
+        }
+
+        /**
+         * Gets the integration id.
+         *
+         * @return the integration id
+         */
+        public Integer getIntegrationId() {
+            return this.integrationId;
+        }
+    }
+
+    /**
+     * Operator parameter.
+     */
+    public static enum Operator {
+        /**
+         * starts_with
+         */
+        STARTS_WITH,
+
+        /**
+         * ends_with
+         */
+        ENDS_WITH,
+
+        /**
+         * contains
+         */
+        CONTAINS,
+
+        /**
+         * regex
+         */
+        REGEX
+    }
+
+    /**
+     * Workflow file reference parameter.
+     */
+    public static class WorkflowFileReference {
+        private String path;
+        private String ref;
+        private int repositoryId;
+        private String sha;
+
+        /**
+         * Instantiates a new workflow file reference.
+         */
+        public WorkflowFileReference() {
```

Comment:
Jackson will do this without a public constructor.  Make it internal  or remove it entirely.